app/presenters/document/web_resource_presenter.rb

```diff
@@ -0,0 +1,175 @@
+module Document
+  ##
+  # Blacklight document presenter for a Europeana web resource
+  class WebResourcePresenter < DocumentPresenter
+    include ActionView::Helpers::NumberHelper
+
+    def initialize(document, record, controller, configuration = controller.blacklight_config)
+      super(document, controller, configuration)
+      @record = record
+      @record_presenter = RecordPresenter.new(record, controller)
+    end
+
+    def media_item
+      {
+        media_type: media_type,
+        rights: simple_rights_label_data,
+        downloadable: downloadable?,
+        playable: playable?,
+        thumbnail: thumbnail,
+        play_url: play_url,
+        technical_metadata: technical_metadata,
+        download: {
+          url: download_url,
+          text: t('site.object.actions.download')
+        }
+      }.tap do |item|
+        if player.nil?
+          item[:is_unknown_type] = url
+        else
+          item[:"is_#{player}"] = true
+        end
+      end
+    end
+
+    def play_url
+      @play_url ||= begin
+        @record_presenter.iiif_manifesto || url
+      end
+    end
+
+    def url
+      @url ||= render_document_show_field_value('about')
+    end
+
+    def media_rights
+      @media_rights ||= begin
+        rights = render_document_show_field_value('webResourceEdmRights')
+        rights.blank? ? @record_presenter.media_rights : rights
+      end
+    end
+
+    def mime_type
+      @mime_type ||= render_document_show_field_value('ebucoreHasMimeType')
+    end
+
+    # Media type function normalises mime types
+    def media_type
```

Comment:
Cyclomatic complexity for media_type is too high. [7/6]<br>Method has too many lines. [18/10]